src/database.rs

```diff
@@ -125,6 +125,82 @@ impl Database {
         Ok(Self { pool })
     }
 
+    pub async fn insert_pending_identity(
+        &self,
+        group_id: usize,
+        identity: &Hash,
+    ) -> Result<(), Error> {
+        self.pool
+            .execute(
+                sqlx::query(
+                    r#"INSERT INTO pending_identities (group_id, commitment)
+                           VALUES ($1, $2);"#,
+                )
+                .bind(group_id as i64)
+                .bind(identity),
+            )
+            .await?;
+        Ok(())
+    }
```

Comment:
Stylistic nitpick: I think it looks better with an intermediate variable:
```rust
        let query = sqlx::query(
            r#"UPDATE pending_identities
                    SET mined_in_block = $1, assigned_leaf_idx = $2
                    WHERE group_id = $3 AND commitment = $4;"#,
        )
        .bind(block_number as i64)
        .bind(tree_idx as i64)
        .bind(group_id as i64)
        .bind(commitment);
        self.pool.execute(query).await?;
        Ok(())
```
This reduces the amount of identation and lines just containing `)`.